### PR TITLE
peer cli assist & integration test in HW mode

### PR DESCRIPTION
**Pull request by g2flyer**
**What this PR does / why we need it**:

This PR provides a small utility which assists the peer cli with message manipulations based on the client sdk code. So far it implements the credential conversion in `initEnclave` and correspondingly simplifies `peer.sh` , makes it work in HW^1 and also ensures what there is a single place which does the actual message manipulation.  It also defines stub for commands for the peer cli to hand off (complicated) client request and response processing.

^1: the major challenge here was that we used inconsistent hex encodings. Took me probably as much time as doing all the rest to figure that out ... :-(

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing changes and/or breaks backward compatability?**:

### CODE REVIEW

File: common/crypto/attestation-api/evidence/verify-evidence.cpp [hunk 1]
```diff
@@ -187,6 +187,8 @@ bool verify_ias_evidence(
                  extract_hex_from_report(ias_report,
                      offsetof(sgx_quote_t, report_body) + offsetof(sgx_report_body_t, mr_enclave),
                      sizeof(sgx_measurement_t), hex_id));
```
Comment (bvavala) [inline common/crypto/attestation-api/evidence/verify-evidence.cpp:189]: should we put here a normalization step (toUpper or to Lower) for both terms?

**Review by bvavala — APPROVED** [review]
Very nice, works like charm.
Just a comment, oth. good to do.

File: common/crypto/attestation-api/evidence/verify-evidence.cpp (hunk 1 above)
Comment (g2flyer) [reply common/crypto/attestation-api/evidence/verify-evidence.cpp:189]: i documented it now in the .md as supposed to be upper-bound.  if we do normalization, i think we would have to do it in peer cli and management API; doing it only here might still bring up issues elsewhere if you confuse cases (e.g., consider org1 installs lowercase, the other one uppercase; from an ERCC case they would now be considered the same but from a ledger perspective they are different!)

Comment (bvavala) [reply common/crypto/attestation-api/evidence/verify-evidence.cpp:189]: I understand, but that upper-case/lower-case encoding is irrelevant -- security-wise -- and just raises implementation issues.
We can discuss that, and I can do the change if we decide so.

Comment (g2flyer) [reply common/crypto/attestation-api/evidence/verify-evidence.cpp:189]: irrelevant security-wise, yes; but functionality-wise it's not as fabric is case-sensitive (at least i thought) for versions.   In that case i don't really see how we could normalize, certainly not transparent to admins? (non-admins should never see the version). But we can talk tomorrow in the call if you want ...

**Pull request merged**